cli/src/main.rs

```diff
@@ -0,0 +1,221 @@
+                            continue;
+                        }
+                        let key = &parts[1];
+                        let value = parts[2..].join(" "); // Support spaces in value
+                        match db_cli.put(key, &value).await {
+                            Ok(_) => println!("Successfully inserted key-value pair."),
+                            Err(e) => eprintln!("Error executing put: {}", e),
+                        }
+                    }
+                    "get" => {
+                        if parts.len() != 2 {
+                            println!("Usage: get <key>");
+                            continue;
+                        }
+                        let key = &parts[1];
+                        match db_cli.get(key).await {
+                            Ok(Some(value)) => println!("{}", value),
+                            Ok(None) => println!("Key not found."),
+                            Err(e) => eprintln!("Error executing get: {}", e),
+                        }
+                    }
+                    "delete" => {
+                        if parts.len() != 2 {
+                            println!("Usage: delete <key>");
+                            continue;
+                        }
+                        let key = &parts[1];
+                        match db_cli.delete(key).await {
+                            Ok(_) => println!("Successfully deleted key."),
+                            Err(e) => eprintln!("Error executing delete: {}", e),
+                        }
+                    }
+                    "help" => {
+                        SlateDbREPL::help();
+                    }
+                    "exit" | "quit" => {
+                        println!("Exiting REPL. Goodbye!");
+                        // Close the database before exiting
+                        if let Err(e) = db_cli.close().await {
+                            eprintln!("Error closing the database: {}", e);
+                        }
+                        break;
+                    }
+                    _ => {
+                        println!(
+                            "Unknown command: '{}'. Type 'help' \
+                        to see available commands.",
+                            command
+                        );
+                    }
+                }
+            }
+            Err(ReadlineError::Interrupted) => {
+                println!("CTRL-C");
+                println!("Exiting REPL. Goodbye!");
+                if let Err(e) = db_cli.close().await {
+                    eprintln!("Error closing the database: {}", e);
+                }
+                break;
+            }
+            Err(ReadlineError::Eof) => {
+                println!("CTRL-D");
+                println!("Exiting REPL. Goodbye!");
+                if let Err(e) = db_cli.close().await {
+                    eprintln!("Error closing the database: {}", e);
+                }
+                break;
```

Comment:
this snippet is used in 3 instances, maybe we can move this to a separate function

Reply:
Actually I just put it at when exiting the loop instead of in the method. Thanks for the tip!